### ISSUE TITLE
LFCC and MFCC for frame by frame processing

### ISSUE THREAD

**Issue by eagomez2** [pytorch/audio#2545]
### 📚 The doc issue

Both ``LFCC`` and ``MFCC`` documentation mention the following:
>This output depends on the maximum value in the input spectrogram, and so may return different values for an audio clip split into snippets vs. a full clip.

Both features need internally to compute the ``Spectrogram`` which has a ``normalized`` argument, although as of now it defaults to ``False`` and therefore, to my understanding there should be no other sources of dependence on the maximum value in the input spectrogram. Additionally, I think if ``center=True`` it may affect a comparison of these features in an audio clip vs the same audio clip split into snippets. Disabling these two, I get an error range somewhere below ``1e-5`` for ``torch.float32`` that I think is expected, but please correct me if I am wrong (see related code [here](https://discuss.pytorch.org/t/mfcc-and-lfcc-in-frame-by-frame-processing/156520)).

In addition to these, ``MelSpectrogram`` method that computes ``Spectrogram`` inside [``MFCC`` implementation](https://pytorch.org/audio/stable/_modules/torchaudio/transforms/_transforms.html#MFCC) does not mention the potential dependence on the maximum value of the input spectrogram in its documentation depending on the ``normalized`` argument that may affect a full clip vs. the same clip into snippets comparison. A same situation occurs with the ``center`` argument.

### Suggest a potential alternative/fix

- If my assumptions are correct, I would expect both ``LFCC`` and ``MFCC`` to mention ``center`` and ``normalized`` arguments as the sources of the potential discrepancy when calculating these features for a full clip vs. the same clip into snippets. I find this information quite useful when porting an offline model to real time.

- Additionally, ``MelSpectrogram`` could include the same information as it also depends internally on ``Spectrogram``

**Comment by nateanl** [comment]
Hi @eagomez2, I checked your code and try to find which line introduces the 1e-5 to 1e-7 error. In [torchaudio.transforms.LFCC](https://github.com/pytorch/audio/blob/main/torchaudio/transforms/_transforms.py#L797), the value difference between full audio solution and frame by frame solution is much less than 1e-7 until this line:
```python
       lfcc = torch.matmul(specgram.transpose(-1, -2), self.dct_mat).transpose(-1, -2)
```
I think the 1e-5 to 1e-7 error is expected as the dtype is `torch.float32`, and most of the bits are used in `self.dct_mat` and `specgram`.

I agree `center` should affect the output if applying `LFCC` frame by frame. If `center` is True, `torch.stft` pads `n_fft / 2` samples on both sides of the waveform. In the whole audio case, the frame in the middle of waveform will not be affected, but in frame by frame case, it is.
For `normalized` parameter, the current implementation doesn't affect the output since it is based on window instead of the spectrogram. 
```python
if normalized: 
     spec_f /= window.pow(2.0).sum().sqrt() 
```
But there is discussion about the implementation (https://github.com/pytorch/audio/issues/2104). We may want to revisit this issue regarding the `normalized` documentation if there is a change in `normalized` implementation.

Best,

**Comment by nateanl** [comment]
Yes, this can be verified by a simple script.
```python
from torchaudio import functional as F
import torch


specgram = torch.rand(1, 200)
specgram = specgram.repeat(100, 1) # make sure each column has the same max value.
dct_mat = F.create_dct(50, 200, "ortho")
whole = torch.matmul(specgram, dct_mat)
frame = torch.matmul(specgram[0:1], dct_mat)
print(whole[0] - frame[0])
```
You can see although `specgram` and `specgram[0:1]` has the same maximum value, there are still errors that are larger than 1e-7 in the outputs. We can make sure the root cause is not from the max value of spectrogram, but rather from the matrix multiplication vs vector multiplication.

**Comment by eagomez2** [comment]
Thanks a lot @nateanl,

Then it's all clear now.
